impl/maven-impl/src/main/java/org/apache/maven/impl/DefaultSourceRoot.java

```diff
@@ -64,30 +64,38 @@ public final class DefaultSourceRoot implements SourceRoot {
      * @param baseDir the base directory for resolving relative paths
      * @param source a source element from the model
      */
+    @SuppressWarnings("checkstyle:AvoidNestedBlocks")
     public DefaultSourceRoot(final Session session, final Path baseDir, final Source source) {
-        String value = nonBlank(source.getDirectory());
-        if (value == null) {
-            throw new IllegalArgumentException("Source declaration without directory value.");
-        }
-        directory = baseDir.resolve(value);
-        FileSystem fs = directory.getFileSystem();
+        FileSystem fs = baseDir.getFileSystem();
         includes = matchers(fs, source.getIncludes());
         excludes = matchers(fs, source.getExcludes());
         stringFiltering = source.isStringFiltering();
         enabled = source.isEnabled();
         moduleName = nonBlank(source.getModule());
-
-        value = nonBlank(source.getScope());
-        scope = (value != null) ? session.requireProjectScope(value) : ProjectScope.MAIN;
-
-        value = nonBlank(source.getLang());
-        language = (value != null) ? session.requireLanguage(value) : Language.JAVA_FAMILY;
-
-        value = nonBlank(source.getTargetVersion());
-        targetVersion = (value != null) ? session.parseVersion(value) : null;
-
-        value = nonBlank(source.getTargetPath());
-        targetPath = (value != null) ? baseDir.resolve(value) : null;
+        {
```

Comment:
No. I am going to -1 this. This is worse than what we had before. Just rename all the variables and don't use the word value, which was a non-descriptive name in the first place, even if it were only used in one place.

Reply:
Reverted to the previous version.